libyul/optimiser/ForLoopConditionIntoBody.h

```diff
@@ -29,17 +30,22 @@ namespace yul
  * By moving the iteration check part into the ForLoop body, we can apply expression splitter
  * to the condition expression.
  *
- * This rewritter will skip loops that already have literal constant as iteration condition.
+ * This rewriter will skip loops that already have literal constant as iteration condition.
  *
  * Requirements:
  * - The Disambiguator must be run upfront.
  * - To avoid unnecessary rewrite, it is recommended to run this rewriter after StructuralSimplifier.
+ * - Only works on EVM code.
```

Comment:
Isn't the point of this PR is to make it also works on eWasm?

Reply:
```suggestion
 * - Only works for dialects with a builtin boolean negation function.
```
?